From: Teemu Ikonen <tpikonen@gmail.com>
Subject: Include glu.h where needed in included structure-synth code.
X-git-branch: p/structuresynth

Also link structuresynth code with libGLU.

This FTBS was exposed by GCC-4.7. See Debian bug #672000 for the same
issue in the real structure-synth Debian package.

 .../ssynth/SyntopiaCore/GLEngine/EngineWidget.h    |    1 +
 .../ssynth/SyntopiaCore/GLEngine/Sphere.h          |    1 +
 .../src/external/structuresynth/structuresynth.pro |    1 +
 3 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/meshlab/src/external/structuresynth/ssynth/SyntopiaCore/GLEngine/EngineWidget.h b/meshlab/src/external/structuresynth/ssynth/SyntopiaCore/GLEngine/EngineWidget.h
index 8905cbe..e13b011 100644
--- a/meshlab/src/external/structuresynth/ssynth/SyntopiaCore/GLEngine/EngineWidget.h
+++ b/meshlab/src/external/structuresynth/ssynth/SyntopiaCore/GLEngine/EngineWidget.h
@@ -5,6 +5,7 @@
 #include <QMainWindow>
 #include <QPoint>
 #include <QList>
+#include <GL/glu.h>
 
 #include "SyntopiaCore/Math/Vector3.h"
 #include "SyntopiaCore/Math/Matrix4.h"
diff --git a/meshlab/src/external/structuresynth/ssynth/SyntopiaCore/GLEngine/Sphere.h b/meshlab/src/external/structuresynth/ssynth/SyntopiaCore/GLEngine/Sphere.h
index 0b0dc40..ed95346 100644
--- a/meshlab/src/external/structuresynth/ssynth/SyntopiaCore/GLEngine/Sphere.h
+++ b/meshlab/src/external/structuresynth/ssynth/SyntopiaCore/GLEngine/Sphere.h
@@ -2,6 +2,7 @@
 
 #include "SyntopiaCore/Math/Vector3.h"
 #include "Object3D.h"
+#include <GL/glu.h>
 
 namespace SyntopiaCore {
 	namespace GLEngine {	
diff --git a/meshlab/src/external/structuresynth/structuresynth.pro b/meshlab/src/external/structuresynth/structuresynth.pro
index d41c607..3e8a881 100644
--- a/meshlab/src/external/structuresynth/structuresynth.pro
+++ b/meshlab/src/external/structuresynth/structuresynth.pro
@@ -120,3 +120,4 @@ win32-msvc2008:DESTDIR       = ../lib/win32-msvc2008
 linux-g++-32:DESTDIR = ../lib/linux-g++-32
 linux-g++-64:DESTDIR = ../lib/linux-g++-64
 linux-g++:DESTDIR = ../lib/linux-g++
+unix:LIBS+=-lGLU
